File: src/2025-02-27-deepclaude37-rust/java/exercises/practice/bottle-song/src/main/java/BottleSong.java
class BottleSong {

    String recite(int startBottles, int takeDown) {
        StringBuilder verses = new StringBuilder();
        for (int i = 0; i < takeDown; i++) {
            int current = startBottles - i;
            if (current <= 0) {
                break;
            }
            int next = current - 1;
            String currentWord = numberToWord(current);
            String nextWord = numberToWord(next);
            
            String line1 = String.format("%s green %s hanging on the wall,", 
                currentWord, current == 1 ? "bottle" : "bottles");
            String line2 = line1;
            String line3 = "And if one green bottle should accidentally fall,";
            String line4 = String.format("There'll be %s green %s hanging on the wall.", 
                nextWord.toLowerCase(), next == 1 ? "bottle" : "bottles");
            
            String verse = String.join("\n", line1, line2, line3, line4);
            if (verses.length() > 0) {
                verses.append("\n\n");
            }
            verses.append(verse);
        }
        return verses.toString();
    }

    private String numberToWord(int n) {
        switch (n) {
            case 10: return "Ten";
            case 9: return "Nine";
            case 8: return "Eight";
            case 7: return "Seven";
            case 6: return "Six";
            case 5: return "Five";
            case 4: return "Four";
            case 3: return "Three";
            case 2: return "Two";
            case 1: return "One";
            case 0: return "no";
            default: throw new IllegalArgumentException("Unsupported number: " + n);
        }
    }

}

File: src/2025-02-27-deepclaude37-rust/javascript/exercises/practice/bottle-song/bottle-song.js
//
// This is only a SKELETON file for the 'Bottle Song' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const recite = (initialBottlesCount, takeDownCount) => {
  const verses = [];
  const numberWords = {
    10: 'Ten',
    9: 'Nine',
    8: 'Eight',
    7: 'Seven',
    6: 'Six',
    5: 'Five',
    4: 'Four',
    3: 'Three',
    2: 'Two',
    1: 'One'
  };
  for (let i = 0; i < takeDownCount; i++) {
    const current = initialBottlesCount - i;
    let currentBottleText, nextBottleText;
    
    // Format current bottle count
    if (current === 1) {
      currentBottleText = 'One green bottle';
    } else {
      currentBottleText = `${numberWords[current]} green bottles`;
    }
    
    // Format next bottle count
    const nextCount = current - 1;
    if (nextCount === 0) {
      nextBottleText = 'no green bottles';
    } else if (nextCount === 1) {
      nextBottleText = 'one green bottle';
    } else {
      nextBottleText = `${numberWords[nextCount].toLowerCase()} green bottles`;
    }
    
    // Construct verse
    const line1 = `${currentBottleText} hanging on the wall,`;
    const line2 = line1;
    const line3 = 'And if one green bottle should accidentally fall,';
    const line4 = `There'll be ${nextBottleText} hanging on the wall.`;
    
    verses.push([line1, line2, line3, line4].join('\n'));
  }
  
  return verses.join('\n\n');
};
